cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -83,36 +85,38 @@
    * @param startTime the start time (inclusive) in the format of "HH:mm", for instance 2am should be "02:00"
    * @param endTime the end time (exclusive) in the format of "HH:mm", for instance 5pm should be "17:00"
    * @param timeZone the time zone of {@code startTime} and {@code endTime}
-   * @return {@link ScheduleConfigurer} containing the time range
+   * @return {@link ScheduleBuilder} containing the time range
    */
-  ScheduleConfigurer setTimeRange(String startTime, String endTime, TimeZone timeZone);
+  ConstraintProgramScheduleBuilder withTimeWindow(String startTime, String endTime, TimeZone timeZone);
 
   /**
    * Set a certain duration passed since the last launching of the Workflow by this schedule,
    * before launching the Workflow again.
    *
    * @param delayMillis duration in milliseconds to wait after the last launching of the Workflow by this schedule
    *                    before launching the Workflow
-   * @return {@link ScheduleConfigurer} containing the given duration
+   * @return {@link ScheduleBuilder} containing the given duration
    */
-  ScheduleConfigurer setDurationSinceLastRun(long delayMillis);
+  ConstraintProgramScheduleBuilder withDurationSinceLastRun(long delayMillis);
 
   /**
    * Create a schedule which is triggered whenever time matching the given cron expression comes.
    * After calling this method, the schedule will be added to the app with all of the information
-   * set in this configurer.
+   * set in this builder.
    *
    * @param cronExpression the cron expression to specify the time to trigger the schedule
+   * @return this {@link ScheduleBuilder}
    */
-  void triggerByTime(String cronExpression);
+  ScheduleCreationSpec triggerByTime(String cronExpression);
```

Comment:
This forces users to call `triggerByTime` after all constraints are set. I feel it will better to still return a `ScheduleBuilder` and have a `build()` method to return `ScheduleCreationSpec`